integrations/azure-devops/azure_devops/webhooks/listeners/push.py

```diff
@@ -27,19 +29,33 @@ async def on_hook(self, data: Dict[str, Any]) -> None:
 
         ref_update_tasks = []
         for update in updates:
-            task = asyncio.create_task(self.process_ref_update(config, update))
+            task = asyncio.create_task(self.process_ref_update(config, update, data))
             ref_update_tasks.append(task)
 
+        logger.debug(f"Created {len(ref_update_tasks)} tasks for processing updates")
+
         await asyncio.gather(*ref_update_tasks, self.register_repository(push_data))
 
     async def process_ref_update(
-        self, config: GitPortAppConfig, update: Dict[str, Any]
+        self,
+        config: GitPortAppConfig,
+        update: Dict[str, Any],
+        initial_data: dict[str, Any] = {},
     ) -> None:
         repo_id = update["repositoryId"]
         branch = extract_branch_name_from_ref(update["name"])
+        default_branch_with_ref: str = initial_data["resource"]["repository"][
+            "defaultBranch"
+        ]
+        default_branch = extract_branch_name_from_ref(default_branch_with_ref)
+        logger.info(
+            f"Processing update for branch {branch}"
+            f" using default branch {default_branch}"
+            f" and config branch {config.branch}"
+        )
         old_commit = update["oldObjectId"]
         new_commit = update["newObjectId"]
-        if config.branch == branch:
+        if branch in [config.branch, default_branch]:
```

Comment:
BTW, currently the default is `main` which I agree is problematic

<img width="944" alt="Screenshot 2024-12-18 at 11 02 40" src="https://github.com/user-attachments/assets/46079888-999f-4514-b05c-052a379451ee" />


I wonder what happens if an existing user had the `main` enabled ( due to default ) and then we change the default to None ( which will mean it should use the default repository branch ), will the previous mapping still have `main` stored in the mapping or will it change to `None`, if it will stay `None` we can just change the default to None.

If it will change to None, it means it will break backwards compatibility (its not something we can allow) and in that case we will want to make the change that will only apply to new users.

Reply:
We want the change to be backwards compatible which means we want to process events from the branch specified in the mapping if it exists. Changing the default to `None` will cause existing user's config to change so it is out of the option.

In the same way for new users, we want to use the default branch. The problem is that there is no way to distinguish new users from old users while maintaining backwards compatibility simply by looking at the branch. What we can do is to process from either branch, hence my implementation.

Reply:
Do you have an implementation in mind to solve this?

Reply:
I would add `useDefaultBranch`

which by default will be `None` and in the new mappings default it will be set to `true`

if `branch` is set and `useDefaultBranch=None | False` then it should use the branch
if `branch` is set and `useDefaultBranch=True` then it should use the default branch of the repository.

what do you think?